contracts/modules/token/TokenController.sol

```diff
@@ -504,10 +596,56 @@ contract TokenController is IERC1132, Iupgradable {
 
     // lock reason removal is skipped here: needs to be done from offchain
 
-    require(token.transfer(_of, _amount));
+    require(token.transfer(_of, _amount), "TokenController: Transfer failed");
     emit Unlocked(_of, _reason, _amount);
   }
 
+  function withdrawCoverNote(
+    address _of,
+    uint[] calldata _coverIds,
+    uint[] calldata _indexes
+  ) external onlyInternal {
+
+    uint reasonCount = lockReason[_of].length;
+    uint lastReasonIndex = reasonCount.sub(1, "TokenController: No locked cover notes found");
+    uint totalAmount = 0;
+
+    // The iteration is done from the last to first to prevent reason indexes from
+    // changing due to the way we delete the items (copy last to current and pop last).
+    // The provided indexes array must be ordered, otherwise reason index checks will fail.
+
+    for (uint i = _coverIds.length; i > 0; i--) {
+
+      bool hasOpenClaim = coverInfo[_coverIds[i - 1]].hasOpenClaim;
+      require(hasOpenClaim == false, "TokenController: Cannot withdraw for cover with an open claim");
+
+      // note: cover owner is implicitly checked using the reason hash
+      bytes32 _reason = keccak256(abi.encodePacked("CN", _of, _coverIds[i - 1]));
+      uint _reasonIndex = _indexes[i - 1];
+      require(lockReason[_of][_reasonIndex] == _reason, "TokenController: Bad reason index");
+
+      uint amount = locked[_of][_reason].amount;
+      require(amount != 0, "TokenController: Locked amount is zero");
```

Comment:
Remove this require so we can bundle zero amount lock reasons when withdrawing rewards for gas saving